pkg/operator/ceph/pool/controller.go

```diff
@@ -149,55 +156,60 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 		logger.Debugf("deleting pool %q", cephBlockPool.Name)
 		err := deletePool(r.context, cephBlockPool)
 		if err != nil {
-			return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to delete pool %q. ", cephBlockPool.Name)
+			return reconcile.Result{}, errors.Wrapf(err, "failed to delete pool %q. ", cephBlockPool.Name)
 		}
 
 		// Remove finalizer
-		err = r.removeFinalizer(cephBlockPool)
+		err = opcontroller.RemoveFinalizer(r.client, cephBlockPool)
 		if err != nil {
-			return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to remove finalizer")
+			return reconcile.Result{}, errors.Wrap(err, "failed to remove finalizer")
 		}
 
 		// Return and do not requeue. Successful deletion.
 		return reconcile.Result{}, nil
 	}
 
-	// CREATE
-	if cephBlockPool.Status == nil || cephBlockPool.Status.Phase != k8sutil.ReadyStatus {
-		reconcileResponse, err := r.reconcileCreatePool(cephBlockPool)
-		if err != nil {
-			return reconcileResponse, errors.Wrapf(err, "failed to create pool %q.", cephBlockPool.GetName())
-		}
+	// Do nothing if reconciling already
+	if cephBlockPool.Status.Phase == k8sutil.ReconcilingStatus {
 		return reconcile.Result{}, nil
```

Comment:
This is done to avoid double reconciliation, (it happens with some events).

Reply:
It will never stay in that state unless the operator dies, but that's more a corner case it'd say.

Reply:
There can be multiple events for reconciling the same object? I would hope we only have a single event for each CR that needs to be reconciled. Maybe it's re-queued, but only one event at a time. If the operator did die, the CR would be stuck in that bad state forever, right? It still seems like the operator should be reconciling here if it's in that state.

Reply:
> There can be multiple events for reconciling the same object?
Somehow that's what I observed.
Also, I've seen other operators doing this check.

> I would hope we only have a single event for each CR that needs to be reconciled. Maybe it's re-queued, but only one event at a time. If the operator did die, the CR would be stuck in that bad state forever, right? It still seems like the operator should be reconciling here if it's in that state.

I guess it's a thread off, double reconciles versus potentially being stuck in that state. This would require the user to edit the object, we should probably don't ask for this...